src/Func.cpp

```diff
@@ -2595,6 +2595,11 @@ vector<Expr> FuncRef::args_with_implicit_vars(const vector<Expr> &e) const {
                 iter++;
             }
         }
+    } else {
+        user_assert(implicit_placeholder_pos == -1)
+          << "Left-hand-side of update definition of " << func.name()
```

Comment:
update definition -> definition

This says "update definition", which is copied from above, but I think it's incorrect here - the message above occurs within a has_pure_definition condition, so you know it's an update definition there. Here it could be either.